sdk/internal/recording/recording.go

```diff
@@ -433,3 +438,221 @@ var modeMap = map[RecordMode]recorder.Mode{
+		recordMode = "playback"
+	}
+	testId := getTestId(pathToRecordings, t)
+
+	url := fmt.Sprintf("%v/%v/start", options.HostScheme(), recordMode)
+
+	req, err := http.NewRequest("POST", url, nil)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("x-recording-file", testId)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	recordingId = resp.Header.Get(IdHeader)
+	return nil
+}
+
+func StopRecording(t *testing.T, options *RecordingOptions) error {
+	defer resetRecordingId()
+
+	if options == nil {
+		options = defaultOptions()
+	}
+
+	url := fmt.Sprintf("%v/%v/stop", options.HostScheme(), recordMode)
+	req, err := http.NewRequest("POST", url, nil)
+	if err != nil {
+		return err
+	}
+	if recordingId == "" {
+		return errors.New("Recording ID was never set. Did you call StartRecording?")
+	}
+	req.Header.Set("x-recording-id", recordingId)
+	_, err = client.Do(req)
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+	return nil
+}
+
+func AddUriSanitizer(replacement, regex string, options *RecordingOptions) error {
+	if options == nil {
+		options = defaultOptions()
+	}
+	url := fmt.Sprintf("%v/Admin/AddSanitizer", options.HostScheme())
+	req, err := http.NewRequest("POST", url, nil)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("x-abstraction-identifier", "UriRegexSanitizer")
+	bodyContent := map[string]string{
+		"value": replacement,
+		"regex": regex,
+	}
+	marshalled, err := json.Marshal(bodyContent)
+	if err != nil {
+		return err
+	}
+	req.Body = ioutil.NopCloser(bytes.NewReader(marshalled))
+	req.ContentLength = int64(len(marshalled))
+	_, err = client.Do(req)
+	return err
+}
+
+func (o *RecordingOptions) Init() {
+	if o.UseHTTPS {
+		o.Host = baseProxyURLSecure
+		o.Scheme = "https"
+	} else {
+		o.Host = baseProxyURL
+		o.Scheme = "http"
+	}
+}
+
+// This looks up an environment variable and if it is not found, returns the recordedValue
+func GetEnvVariable(t *testing.T, varName string, recordedValue string) string {
+	val, ok := os.LookupEnv(varName)
+	if !ok {
+		t.Logf("Could not find environment variable: %v", varName)
+		return recordedValue
+	}
+	return val
+}
+
+func LiveOnly(t *testing.T) {
+	if GetRecordMode() != modeRecording {
+		t.Skip("Live Test Only")
+	}
+}
+
+// Function for sleeping during a test for `duration` seconds. This method will only execute when
+// AZURE_RECORD_MODE = "record", if a test is running in playback this will be a noop.
+func Sleep(duration int) {
+	if GetRecordMode() == modeRecording {
+		time.Sleep(time.Duration(duration) * time.Second)
+	}
+}
+
+func GetRecordingId() string {
+	return recordingId
+}
+
+func GetRecordMode() string {
+	return recordMode
+}
+
+func InPlayback() bool {
+	return GetRecordMode() == modePlayback
+}
+
+func InRecord() bool {
+	return GetRecordMode() == modeRecording
+}
+
+func getRootCas() (*x509.CertPool, error) {
+	localFile, ok := os.LookupEnv("PROXY_CERT")
+
+	rootCAs, err := x509.SystemCertPool()
+	if err != nil {
```

Comment:
Might not be a huge deal but is there a specific error that you're ignoring here (like system cert pool not found)? Or is it literally _any_ error?

If there is a specific one it'd be good to just check that, and return the error otherwise (perhaps there's something dire going on).